internal-shared/protomappers/mappers.go

```diff
@@ -347,3 +349,15 @@ func LogViewerProto(
 
 	return out
 }
+
+// This gets called on the plugin side on the way _in_ to convert back to a component
+// Only exists to satisfy argmapper - the value isn't used.
+func DeclaredResourcesResp(input *pb.DeclaredResources) *component.DeclaredResourcesResp {
+	return &component.DeclaredResourcesResp{}
+}
+
+// Gets called from core side to serialize the outparameter
+// This value also doesn't really end up getting used.
+func DeclaredResourcesRespProto(input *component.DeclaredResourcesResp) *pb.DeclaredResources {
+	return &pb.DeclaredResources{}
+}
```

Comment:
So because `component.DeclaredResourceResp` never crosses the plugin boundary now (as it is only seen on the plugin side to be populated) we can just remove this code now.

Reply:
For argmapper on the core side to be able to call the plugin function (below), it needs to believe it has all of the arguments the DeployFunc is looking for. Without these mappers, it doesn't think it can satisfy the argument and fails.

https://github.com/hashicorp/waypoint/pull/1737/files#diff-fea171fe74bf37c5196be1edbcc06902a4821ae56acea2613983cb56e99a76cbR327

Reply:
Well, because the above code isn't actually populating the return values and your new code sees return data, that's a giveaway to me that this code isn't being used.

Reply:
It gets called, I promise! Without a mapper from the proto to the component, the plugin can't generate a proto-only version of the DeployFunc, and it fails here: https://github.com/hashicorp/waypoint-plugin-sdk/blob/main/internal/funcspec/spec.go#L50

And then we need the component -> proto mapper on the core side to generate the required arguments.

Maybe there's some clever way to make mappers to/from nil or empty named messages to make it more clear these values aren't used?

Reply:
If it's being called, the value that is convert isn't actually used, right? So we should be able to remove even passing it in.

Reply:
I've removed these in favor of a new filter function that prevents the outparameter from making its way into the advertised function spec: https://github.com/hashicorp/waypoint-plugin-sdk/pull/40/files#diff-6e083c58b681306ffed3d828652a8e0e068bcc3e4708ba40f1db6e355906a6dfR29